src/client/websocket/WebSocketManager.js

```diff
@@ -79,6 +85,21 @@ class WebSocketManager {
     this.client.emit(Events.DEBUG, `[connection] ${message}`);
   }
 
+  async _handleSessionLimit(shard) {
+    if (!this.sessionStartLimit && this.spawnQueue.length) {
```

Comment:
I am not sure what the intent of this method is.
Ignore the session start limit if the queue is not empty and no limit is available and spawn?
Fetch the limit, wait if necessary, and spawn if one is available?